external-builds/pytorch/patches/comfyui/v0.3.41/comfyui/base/0001-therock-install-and-launch-scripts-for-comfyui.patch

```diff
@@ -0,0 +1,70 @@
+From 77450953af8a3d3080f716c08e0fb9295fb9098b Mon Sep 17 00:00:00 2001
+From: Mika Laitio <mika.laitio@amd.com>
+Date: Tue, 10 Jun 2025 17:37:18 -0700
+Subject: [PATCH] therock install and launch scripts for comfyui
+
+1. thr_install.sh is invoked by the rockbuilders
+  comfyui.cfg to install the comfyui to directory
+  in $ROCM_HOME/apps/comfyui
+  This script will take two parameters:
+  - location containing the git clone of comfyui sources
+  - ROCM_HOME directory
+
+2. thr_comfyui.sh is the launcher script that will start
+  the comfyui from it's install location
+  ($ROCM_HOME/apps/comfyui)
+
+Signed-off-by: Mika Laitio <mika.laitio@amd.com>
+---
+ thr_comfyui.sh | 11 +++++++++++
+ thr_install.sh | 20 ++++++++++++++++++++
+ 2 files changed, 31 insertions(+)
+ create mode 100644 thr_comfyui.sh
+ create mode 100755 thr_install.sh
+
+diff --git a/thr_comfyui.sh b/thr_comfyui.sh
+new file mode 100644
+index 00000000..e5a503b8
+--- /dev/null
++++ b/thr_comfyui.sh
+@@ -0,0 +1,11 @@
++if [ -z "$ROCM_HOME" ]; then
++    echo "Error, ROCM_HOME not specified."
++    exit 1
++fi
++if test -d "${ROCM_HOME}/apps/comfyui"; then
++    cd ${ROCM_HOME}/apps/comfyui
++    export LD_LIBRARY_PATH=${ROCM_HOME}/lib:${ROCM_HOME}/lib/llvm/lib:${LD_LIBRARY_PATH}
++    python main.py
```

Comment:
Shouldn't need to override LD_LIBRARY_PATH. That needs to come from "installing" ROCm, either via tarballs or pip wheels and the `rocm*` packages.

With that resolved, there is nothing left in this script but `python main.py`, at which point I'm wondering why we even need a build configuration for this project at all.

Let's check how PyTorch uses the rocm wheels to see if similar changes to comfyui would be needed for compatibility.

Reply:
Actually, https://github.com/comfyanonymous/ComfyUI?tab=readme-ov-file#amd-gpus-linux-only maybe all we need is to test with our torch wheels: https://github.com/ROCm/TheRock/blob/main/RELEASES.md#installing-pytorch-python-packages. With a quick look through ComfuUI, I don't see anywhere using ROCm directly, just via torch.

Reply:
Python projects may want to preload using the rocm_sdk package, as we do in PyTorch: https://github.com/ROCm/TheRock/blob/main/external-builds/pytorch/patches/pytorch/v2.7.0/pytorch/hipified/0015-Preload-rocm-sdk-binaries-if-available.patch . That can substitute for `LD_LIBRARY_PATH` or `LD_PRELOAD` environment variable tricks. I think in this case though, we just need to test upstream comfyui with the torch wheels that we publish.